fontdrasil/src/types.rs

```diff
@@ -156,4 +157,18 @@ impl WidthClass {
             WidthClass::UltraExpanded => 200.0,
         }
     }
+
+    pub fn from_percent(percent: f32) -> Self {
+        let mappings = WidthClass::all_values()
+            .iter()
+            .map(|wc| (wc.to_percent().into(), (*wc as u16).into()))
+            .collect();
+        let v: u16 = PiecewiseLinearMap::new(mappings)
```

Comment:
I don't think I'd have expected this to allocate. 

This is testable, I suggest adding tests including out of bounds values. At a glance I'm slightly skeptical this works if the value if out of bounds.

Reply:
I tried it, `WidthClass::from_percent(42.0)` fails (bonus points: not the way I expected  ) with `attempt to subtract with overflow`

Reply:
Over-large values also fail.

Reply:
Exploratory fix (PR to this branch) in https://github.com/googlefonts/fontc/pull/923

Reply:
but why doesn't it work? I thought PiecewiseLinearMap worked the same way as fonttools' equivalent. I used the same function that varLib uses for the same task and naively assumed they'd work the same. Should they not work the same?

Reply:
Fair point, the non-handling of our of bounds values should be viewed as a bug then. I will file an issue. I still think we prefer not to allocate here.

Reply:
https://github.com/googlefonts/fontc/issues/924